roles/core/dns_server/tasks/main.yml

```diff
@@ -26,6 +26,27 @@
   tags:
     - package
 
+# Master DNS is the dns_ip of primary management network
+# (management_network_naming + '1-1')
+- name: Set master DNS server - Get DNS IP
+  set_fact:
+    dns_master: "{{ networks[management_network_naming+'1-1'].services_ip.dns_ip }}"
+  when:
+    - "dns_master is not defined"
+    - "networks[j2_node_main_network].services_ip.dns_ip is defined"
+
+- name: Assert dns_master is defined
+  assert:
+    that:
+      - dns_master is defined
+    fail_msg: "'dns_master' must be defined"
```

Comment:
I think this message is not verbose enough. Wouldn't it be better to say :
`"Could not get a dns_master. Please ensure that dns_master variable is defined, or that networks[{{j2_node_main_network}}].services_ip.dns_ip is defined and contains a valid ip or list of ip."`

Reply:
Is this better?
```
fatal: [mgmt0-master]: FAILED! => {
    "assertion": "dns_master is defined", 
    "changed": false, 
    "evaluated_to": false, 
    "msg": "'dns_master' must be defined. Please ensure that either 'dns_master' or networks[ice1-1].services_ip.dns_ip is defined and contains a valid IP address."
}
```

Reply:
Done